mario-karen/server.js

```diff
@@ -0,0 +1,184 @@
+  ]
+
+  client.query( SQL, values )
+    .then(function() {
+      response.send('insert complete')
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.put('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // 2345 with CRUD as Update. interacting with updateRecord.
+
+  let SQL = 'UPDATE articles SET title = $2, author =$3, "authorUrl" = $4, category = $5, "publishedOn" = $6, body = $7 WHERE article_id=$1;';
+  let values = [
+    request.params.id,
+    request.body.title,
+    request.body.author,
+    request.body.authorUrl,
+    request.body.category,
+    request.body.publishedOn,
+    request.body.body
+  ];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('update complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // 2345 CRUD is Destroy interacting with deleteRecord in article.js
+
+  let SQL = `DELETE FROM articles WHERE article_id=$1;`;
+  let values = [request.params.id];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('Delete complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // 2345 CRUD is DESTROY and interacting with truncateTable at article.js
+
+  let SQL = 'DROP TABLE articles;';
+  client.query( SQL )
+    .then(() => {
+      response.send('Delete complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+// COMMENT: What is this function invocation doing?
+// LoadDB creates a database and filling up that table if it already isnt there.
+loadDB();
+
+app.listen(PORT, () => {
+  console.log(`Server started on port ${PORT}!`);
+});
+
+
+//////// ** DATABASE LOADER ** ////////
+////////////////////////////////////////
+function loadArticles() {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // CRUD is CREATE, 3,4 if DB has no values in it 2,3
```

Comment:
This is pretty independent from CRUD and our normal database operations; this is just what happens when the server is started up, initially.